### ISSUE TITLE
Support annotations defined in docstrings

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#839]
In addition to Python 3 function annotations and type comments we should also support annotations in docstrings, following Google docstring conventions. Example:

```python
def f(x):
    """description...

    Args:
        x (int): describe...

    Returns:
        List[str]: describe...
    """
    ...

f('x')   # Error
```

This would be primarily useful for code that needs to be Python 2 compatible. It should be possible to mix and match different kinds of annotations in a file. A type comment should take precedence over a docstring.

This should probably be opt-in (per file) so that we wouldn't give false positives for docstrings that don't fully conform to the expected syntax.

**Comment by presidento** [comment]
There was a similiar issue for this before: #612

**Comment by JukkaL** [comment]
This partially addresses #612, but I'll leave #612 still open since we may wish to support additional conventions.

**Comment by JukkaL** [comment]
As discussed in #612, I suppose we aren't going to implement this and we can close this.
